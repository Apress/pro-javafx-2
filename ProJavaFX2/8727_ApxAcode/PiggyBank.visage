/*
 * @test
 * @run
 */

import java.lang.*;

class PiggyBank {
  var name: String;
  var pennies: Integer;
  var nickels: Integer;

  function putInPennies(count: Integer):Void {
    if (count <= 0) {
      throw new IllegalArgumentException("count must be positive.");
    } else {
      pennies += count;
      println("You put {count} pennies into {name}.");
    }
  }

  function putInNickels(count: Integer):Void {
    if (count <= 0) {
      throw new IllegalArgumentException("count must be positive.");
    } else {
      nickels += count;
      println("You put {count} nickels into {name}.");
    }
  }

  function total():Integer {
    pennies + nickels * 5
  }
}

var myPiggyBank = PiggyBank { name: "My Piggy Bank" };
myPiggyBank.putInPennies(7);
myPiggyBank.putInNickels(3);
myPiggyBank.putInNickels(1);
println("{myPiggyBank.name} has {myPiggyBank.total()} cents:  "
  "{myPiggyBank.pennies} pennies and {myPiggyBank.nickels} nickels.");

var yourPiggyBank = PiggyBank { name: "Your Piggy Bank" };
yourPiggyBank.putInPennies(4);
yourPiggyBank.putInNickels(6);
yourPiggyBank.putInPennies(9);
println("{yourPiggyBank.name} has {yourPiggyBank.total()} cents:  "
  "{yourPiggyBank.pennies} pennies and {yourPiggyBank.nickels} nickels.");
