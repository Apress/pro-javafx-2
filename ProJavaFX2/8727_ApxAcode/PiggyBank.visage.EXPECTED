You put 7 pennies into My Piggy Bank.
You put 3 nickels into My Piggy Bank.
You put 1 nickels into My Piggy Bank.
My Piggy Bank has 27 cents:  7 pennies and 4 nickels.
You put 4 pennies into Your Piggy Bank.
You put 6 nickels into Your Piggy Bank.
You put 9 pennies into Your Piggy Bank.
Your Piggy Bank has 43 cents:  13 pennies and 6 nickels.
